### PR TITLE
[3.12.x] Try harder to kill agent process in case of 'agent_expireafter' in cf-execd

### PR DESCRIPTION
**Pull request by vpodzime**
3.12.x version of #3935

### CODE REVIEW

**Comment by vpodzime** [comment]
@cf-bottom jenkins please

**Comment by cf-bottom** [comment]
Alright, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4201/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4201/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4201/

File: cf-execd/cf-execd-runner.c [hunk 1]
```diff
@@ -262,11 +263,53 @@ void LocalExec(const ExecConfig *config)
...
+
+                /* We need to determine the PID of the agent (and thus its
+                 * process group) first.*/
+                ClearProcessTable();
+                if (LoadProcessTable())
+                {
+                    ProcessSelect ps = PROCESS_SELECT_INIT;
+                    ps.min_ppid = pid_shell;
+                    ps.max_ppid = pid_shell;
+                    Item *procs = SelectProcesses(".*" /* any command */, &ps, true /* apply ps */);
```
Comment (olehermanse) [inline cf-execd/cf-execd-runner.c]: SelectProcesses takes ps by value, not by pointer, at least in current 3.12.x

**Comment by vpodzime** [comment]
@cf-bottom jenkins, please

**Comment by cf-bottom** [comment]
Sure, I triggered a build:

[Build Status](https://ci.cfengine.com//job/pr-pipeline/4208/)

**Jenkins:** https://ci.cfengine.com/job/pr-pipeline/4208/

**Packages:** http://buildcache.cfengine.com/packages/testing-pr/jenkins-pr-pipeline-4208/

**Pull request merged**